src/test/regress/sql/pg16.sql

```diff
@@ -332,6 +332,37 @@ SELECT pg_get_viewdef('pg16.prop_view_1', true);
 \c - - - :master_port
 SET search_path TO pg16;
 
+--
+-- PG16 allows GRANT WITH ADMIN | INHERIT | SET
+--
+-- GRANT privileges to a role or roles  
+\c - - - :master_port
+CREATE ROLE create_role;
+CREATE ROLE create_role_2;
+CREATE ROLE create_role_3;
+CREATE ROLE create_role_4;
+CREATE USER create_user;
+CREATE USER create_user_2;
+CREATE GROUP create_group;
+CREATE GROUP create_group_2;
+
+--test grant role
+GRANT create_group TO create_role;
+GRANT create_group TO create_role_2 WITH ADMIN OPTION;
+GRANT create_group TO create_role_3 WITH INHERIT;
+GRANT create_group TO create_role_4 WITH SET; 
+
+-- ADMIN role can perfom administrative tasks 
+-- role can now access the data and permissions of the table (owner of table)
+-- role can change current user to any other user/role that has access 
+GRANT ADMIN TO joe;
+GRANT INHERIT ON ROLE joe TO james;
+
+GRANT SELECT ON companies TO joe WITH GRANT OPTION;
+GRANT SET (SELECT) ON companies TO james;
+
```

Comment:
In the tests we are mostly interested in the GRANT command's propagation. Hence, we should check the workers. I suggest you connect to one of the workers and see whether the inherit/set options have been propagated correctly. You can do that by the following commands:
```SQL
\c - - - :worker_1_port
SELECT roleid::regrole::text AS role, member::regrole::text, inherit_option, set_option FROM pg_auth_members
```

Reply:
Still working on this and `pg16.out`. Need a bit more clarification and understanding